src/ORM/Association/ExternalAssociationTrait.php

```diff
@@ -57,7 +57,7 @@ public function foreignKey($key = null)
     {
         if ($key === null) {
             if ($this->_foreignKey === null) {
-                $this->_foreignKey = $this->_modelKey($this->source()->alias());
+                $this->_foreignKey = $this->_modelKey($this->source()->table());
```

Comment:
I doubt it will cause BC issues. The fact that alias is used is actually causing problems. By default the alias is same as table name.